### PR TITLE
Issue CORE-303 | Handle authentication failure

### PR DESCRIPTION
**Pull request by nicholaspcr**
# Description

Kind: issue

Section: cmd/inspr/cli

Summary: changing the message shown when an apply operation gets the http request return of 401 or 403.

Issue link: https://inspr.atlassian.net/jira/software/projects/CORE/boards/44?selectedIssue=CORE-303

## Changelog

> List the main points of change

- cli-client tests
- handleResponseErr -> in the request.go file
- apply.go tests

## Pay attention to
- request.go
- apply.go

> Important parts of the code that require special attention

### CODE REVIEW

File: cmd/inspr/cli/utils/cli_client.go [hunk 1]
```diff
@@ -61,3 +63,22 @@ func SetClient(url string) {
...
+		switch ierr.Code {
+		case ierrors.Unauthorized:
+			fmt.Fprintf(w, "Did you login ?\n")
+		case ierrors.Forbidden:
+			fmt.Fprintf(w, "Forbidden operation, please check for the scope.\n")
+		default:
+			fmt.Fprintf(w, "unexpected inspr error, the message is: %v\n", err.Error())
+		}
+	} else {
+		fmt.Fprintf(w, "Non inspr error, the message is: %v\n", err.Error())
```
Comment (franciscobonand) [inline cmd/inspr/cli/utils/cli_client.go]: Capitalized errors

**Review by franciscobonand — CHANGES_REQUESTED** [review]
The authentication failure handler for the other commands (delete, get, describe) needs to be done.
Also, pull the changes done by the PR of issue CORE-307

File: cmd/inspr/cli/utils/cli_client.go (hunk 1 above)
Comment (nicholaspcr) [reply cmd/inspr/cli/utils/cli_client.go]: fixed

File: cmd/inspr/cli/apply_test.go [hunk 2]
```diff
@@ -312,25 +321,80 @@ func Test_applyValidFiles(t *testing.T) {
...
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			out := &bytes.Buffer{}
-			GetFactory().Subscribe(meta.Component{
-				APIVersion: "v1",
-				Kind:       "dapp",
-			},
+			applyFactory = nil
```
Comment (LucasStarlingdePaulaSalles) [inline cmd/inspr/cli/apply_test.go:371]: Is this been used?

File: pkg/rest/request/request.go [hunk 3]
```diff
@@ -86,20 +91,34 @@ type Client struct {
...
+			return err
+		}
+		return defaultErr
+	case http.StatusForbidden:
+		decoder.Decode(&err)
+		if err != nil {
+			err.Wrap("status forbidden")
+			return err
+		}
+		return defaultErr
```
Comment (LucasStarlingdePaulaSalles) [inline pkg/rest/request/request.go]: This will conflict with develop, merge it to your branch

**Review by LucasStarlingdePaulaSalles — CHANGES_REQUESTED** [review]

**Review by franciscobonand — APPROVED** [review]

**Review by LucasStarlingdePaulaSalles — APPROVED** [review]

**Pull request merged**